### PR TITLE
[SPARK-27794][R][DOCS][BACKPORT] Use https URL for CRAN repo

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Use https URL for CRAN repo (and for a Scala download in a Dockerfile)
Backport of https://github.com/apache/spark/pull/24664

## How was this patch tested?

Existing tests

### CODE REVIEW

File: dev/create-release/spark-rm/Dockerfile [hunk 1]
```diff
@@ -38,7 +38,8 @@ ARG PIP_PKGS="pyopenssl pypandoc numpy pygments sphinx"
 #
 # This is all in a single "RUN" command so that if anything changes, "apt update" is run to fetch
 # the most current package versions (instead of potentially using old versions cached by docker).
-RUN echo 'deb http://cran.cnr.Berkeley.edu/bin/linux/ubuntu xenial/' >> /etc/apt/sources.list && \
+RUN apt-get clean && apt-get update && $APT_INSTALL gnupg ca-certificates apt-transport-https && \
```
Comment (srowen) [inline dev/create-release/spark-rm/Dockerfile]: This also needed `apt-transport-https` in Ubuntu 16, it seems. I tried the steps in Docker and it successfully accesses CRAN.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. I also tested with docker.
Thank you, @srowen !

**Comment by dongjoon-hyun** [comment]
This is for both `branch-2.4/2.3`?

**Comment by srowen** [comment]
I'm not sure if it back-ports cleanly to 2.3, but it may well. It can be back-ported to 2.3 safely I think.

**Comment by dongjoon-hyun** [comment]
Merged to branch-2.4.
`branch-2.3` was impossible because `dev/create-release/spark-rm` was created at `branch-2.4`.

**Pull request closed without merging**